File: order-tracker/lib/features/orders/domain/entities/order_summary.dart
import 'package:equatable/equatable.dart';
import 'package:order_tracker/core/money/money.dart';
import 'package:order_tracker/features/orders/domain/entities/market_code.dart';
import 'package:order_tracker/features/orders/domain/entities/order_status.dart';

final class OrderSummary extends Equatable {
  const OrderSummary({
    required this.orderId,
    required this.status,
    required this.market,
    required this.clientId,
    required this.grandTotal,
    required this.processedAt,
    this.eventVersion,
    this.reason,
  });

  final String orderId;
  final OrderStatus status;
  final MarketCode market;
  final String clientId;
  final int? eventVersion;
  final Money grandTotal;
  final String? reason;
  final DateTime processedAt;

  @override
  List<Object?> get props => [
    orderId,
    status,
    market,
    clientId,
    eventVersion,
    grandTotal,
    reason,
    processedAt,
  ];
}

File: order-processor/src/main/java/com/grupomariposa/orders/application/query/OrderSummary.java
package com.grupomariposa.orders.application.query;

import com.grupomariposa.orders.domain.model.CurrencyCode;
import com.grupomariposa.orders.domain.model.MarketCode;
import com.grupomariposa.orders.domain.model.Money;
import com.grupomariposa.orders.domain.model.OrderStatus;
import com.grupomariposa.orders.domain.model.RejectionCode;
import java.time.Instant;

public record OrderSummary(
        String orderId,
        OrderStatus status,
        MarketCode market,
        CurrencyCode currency,
        String clientId,
        int eventVersion,
        Money grandTotal,
        RejectionCode reason,
        Instant processedAt) {
}
